### PR TITLE
Fully cover remaining asserts

### PR DESCRIPTION
**Pull request by barnjamin**

### CODE REVIEW

File: examples/amm/amm.py [hunk 1]
```diff
@@ -360,7 +358,6 @@ def burn(
                 pool_bal.hasValue(),
                 a_bal.hasValue(),
                 b_bal.hasValue(),
-                comment=ConstantProductAMMErrors.MissingBalances,
```
Comment (michaeldiamant) [inline examples/amm/amm.py:363]: @barnjamin For my understanding:  In a few places, the PR removes `comment=MissingBalances`.  Can you help me understand the rationale?

File: examples/amm/amm_test.py [hunk 2]
```diff
@@ -673,20 +735,73 @@ def burn(
...
+            fake_client,
+        )
+
+        return well_formed_swap + valid_swap_xfer
 
     all_asserts: dict[int, ProgramAssertion] = creator_app_client.approval_asserts  # type: ignore[assignment]
-    for msg, method, kwargs in bootstrap_cases() + mint_cases() + burn_cases():
-        print(f"Testing {msg}")
+
+    print(f"Total asserts ({len(all_asserts)})")
```
Comment (michaeldiamant) [inline examples/amm/amm_test.py:793]: @barnjamin Minor and if you feel it's out of scope, I'm on-board to approve as is.  Let me know what you think?

I think it'd be nice to factor out 2 tests that use all test cases as input (naming can be modified):
* `test_asserts` - Confirms that each case raises an assertion.  Helps user understand when assertion test coverage regresses.
* `test_assert_coverage` - Fails when > 0 assertions are _not_ tested.  Helps user understand when assertion test coverage is partial.

I tried to do accomplish the goal locally, but I'm unable to make it work.  I think the idea is to return 
```   
        set_governor_cases()
        + bootstrap_cases()
        + mint_cases()
        + burn_cases()
        + swap_cases()
```

The challenge I ran into is combining `@pytest.mark.parameterize` with `@pytest.fixture`.  The parameterize input requires fixture values as input, and I was unable to find a way to make it work.  

I can also explain live if helpful.  I'm sure there's a way and it must be my lack of pytest awareness that's holding me up.

File: examples/amm/amm.py (hunk 1 above)
Comment (barnjamin) [reply examples/amm/amm.py:363]: The comments are useful for conveying a reason to the caller. They also do double duty by flagging an assert as one we'd want to test. In this case, I don't see them as useful to tell the caller about (not to mention tough to test)

Comment (michaeldiamant) [reply examples/amm/amm.py:363]: @barnjamin Got it - works for me.  Resolving.

Out of band to the PR, I think it's worth _some_ consideration to see if we can identify a strategy to more easily test these more difficult-to-test states.

**Pull request merged**